service/history/handler.go

```diff
@@ -157,7 +158,12 @@ var (
 )
 
 // NewHandler creates a thrift handler for the history service
-func NewHandler(args NewHandlerArgs) *Handler {
+func NewHandler(args NewHandlerArgs) (*Handler, error) {
+	sdkSystemClient, err := args.SdkClientFactory.NewSystemClient(args.Logger)
```

Comment:
why not extract this into provider and inject SystemSdkClient?

Reply:
Fixed!